source/server/BUILD

```diff
@@ -20,6 +20,10 @@ envoy_cc_library(
     name = "configuration_lib",
     srcs = ["configuration_impl.cc"],
     hdrs = ["configuration_impl.h"],
+    copts = select({
+        "//bazel:disable_lightstep_tracing": ["-DDISABLE_LIGHTSTEP_TRACING"],
```

Comment:
nit: This is where I would put the hack TODO comment. If we do registration correctly, we should not needed any defines. We are going to have to have selectors no matter what to compile different code.